gcp_variant_transforms/libs/bigquery_util.py

```diff
@@ -360,3 +375,115 @@ def calculate_optimal_partition_size(total_base_pairs):
+    load_job = self._client.load_table_from_uri(
+        uri, table_id, job_config=job_config)
+    self._suffixes_to_load_jobs.update({suffix: load_job})
+
+  def _cancel_all_running_load_jobs(self):
+    for load_job in self._suffixes_to_load_jobs.values():
+      load_job.cancel()
+
+  def _handle_failed_load_job(self, suffix, load_job):
+    if self._num_load_jobs_retries < _BQ_LOAD_JOB_NUM_RETRIES:
+      self._num_load_jobs_retries += 1
+      # Retry the failed job after 5 minutes wait.
+      time.sleep(300)
+      self._start_one_load_job(suffix)
+    else:
+      # Jobs have failed more than _BQ_LOAD_JOB_NUM_RETRIES, cancel all jobs.
+      self._cancel_all_running_load_jobs()
+      table_id = compose_table_name(self._table_base_name, suffix)
+      raise ValueError(
+          'Failed to load AVRO to BigQuery table {} \n state: {} \n '
+          'job_id: {} \n errors: {}.'.format(table_id, load_job.state,
+                                             load_job.path,
+                                             '\n'.join(load_job.errors)))
+  def _monitor_load_jobs(self):
+    # Waits until current jobs are done and then add remaining jobs one by one.
+    while self._suffixes_to_load_jobs:
+      time.sleep(60)
+      processed_suffixes = self._suffixes_to_load_jobs.keys()
+      for suffix in processed_suffixes:
+        load_job = self._suffixes_to_load_jobs.get(suffix)
+        if load_job.done():
+          del self._suffixes_to_load_jobs[suffix]
+          if load_job.state != 'DONE':
+            self._handle_failed_load_job(suffix, load_job)
+          else:
+            if self._remaining_load_jobs:
+              next_suffix = self._remaining_load_jobs.pop()
+              self._start_one_load_job(next_suffix)
+
+
+def create_output_table(full_table_id, total_base_pairs, schema_file_path):
+  # type: (str, int, str) -> None
+  """Creates an integer range partitioned table using `bq mk table...` command.
+
+  Since beam.io.BigQuerySink is unable to create an integer range partition
+  we use `bq mk table...` to achieve this goal. Note that this command runs on
+  the worker that monitors the Dataflow job.
+
+  Args:
+    full_table_id: for example: projet:dataset.table_base_name__chr1
+    total_base_pairs: the maximum expected value of `start_position` column.
+    schema_file_path: a json file that contains the schema of the table.
+  """
+  (partition_size, total_base_pairs_enlarged) = (
+      calculate_optimal_partition_size(total_base_pairs))
+  bq_command = _BQ_CREATE_PARTITIONED_TABLE_COMMAND.format(
+      TOTAL_BASE_PAIRS=total_base_pairs_enlarged,
+      PARTITION_SIZE=partition_size,
+      FULL_TABLE_ID=full_table_id,
+      SCHEMA_FILE_PATH=schema_file_path)
+  result = os.system(bq_command)
+  if result != 0:
+    time.sleep(30)  # In our integration tests sometime we overwhelm BQ server.
+    result_second_attempt = os.system(bq_command)
+    if result_second_attempt != 0:
```

Comment:
I added an extra re-try to avoid cases where our integration tests fail due to this error:
```
BigQuery error in mk operation: Got 503 response from discovery url:
https://www.googleapis.com/discovery/v1/apis/bigquery/v2/rest
```
As soon as we switch to using BigQuery Python API for creating tables we will remove this temporary solution.